src/main/scala/com.socrata.geoexport/encoders/ShapefileEncoder.scala

```diff
@@ -0,0 +1,201 @@
+
+object ShapefileEncoder extends GeoEncoder {
+  case class FeatureCollectionException(val message: String) extends Exception
+
+  type SoQLColumn = (String, SoQLType)
+  type IntermediarySchema = Seq[ShapeRep[_ <: SoQLValue]]
+  type IntermediaryValues = Seq[(_ <: SoQLValue, ShapeRep[_ <: SoQLValue])]
+
+  val shapefileExt = "shp"
+  val spatialIndexExt = "shx"
+  val dbaseExt = "dbf"
+  val shpFixExt = "fix"
+  val projExt = "prj"
+
+  val shapeArchiveExts = Seq(
+    shapefileExt,
+    spatialIndexExt,
+    dbaseExt,
+    shpFixExt,
+    projExt
+  )
+  private def getShapefileMinions(shpFile: File): Seq[File] = {
+    shpFile.getName.split("\\.") match {
+      case Array(name, ext) => shapeArchiveExts.map { other =>
+        new File(shpFile.getParent, s"${name}.${other}")
+      }
+    }
+  }
+
+  private def toIntermediaryReps(schema: Seq[SoQLColumn]) = schema.map(ShapeRep.repFor(_, ShapefileRepMapper))
+
+  private def buildFeatureType(schema: IntermediarySchema): SimpleFeatureType = {
+    val builder = new SimpleFeatureTypeBuilder()
+
+    builder.setCRS(DefaultGeographicCRS.WGS84)
+    builder.setName("FeatureType")
+
+    val addedNames = schema
+    .filter {
+      case r: IDRep => false
+      case _ => true
+    }.flatMap { rep =>
+      val restrictions = seqAsJavaList(List[Filter]())
+
+      val attrNames = rep.toAttrNames
+      val bindings = rep.toAttrBindings
+
+      if(attrNames.size != bindings.size) {
+        throw new Exception("Attribute names:bindings mapping must be 1:1 in the intermediary representation")
+      }
+
+      attrNames.zip(bindings).foreach { case (attrName, binding) =>
+        // geotools is dumb
+        // scalastyle:off
+        if (classOf[Geometry].isAssignableFrom(binding)) {
+          val attrName = new NameImpl("the_geom")
+          val geomType = new GeometryTypeImpl(attrName, binding, DefaultGeographicCRS.WGS84, true, false, seqAsJavaList(List()), null, null)
+          val geomDescriptor = new GeometryDescriptorImpl(geomType, attrName, Int.MinValue, Int.MaxValue, true, null)
+          builder.add(geomDescriptor)
+        } else {
+          val bindingName = new NameImpl(binding.toString())
+          val attrType = new AttributeTypeImpl(bindingName, binding, true, false, restrictions, null, null)
+          val descriptor = new AttributeDescriptorImpl(attrType, new NameImpl(attrName), Int.MinValue, Int.MaxValue, true, null)
+          builder.add(descriptor)
+        }
+        // scalastyle:on
+      }
+      attrNames
+    }
+    log.debug(s"Added names ${addedNames.size}  ${addedNames}")
+    builder.buildFeatureType()
+  }
+
+  // scalastyle:off
+  private def buildFeature(
+    featureId: Int,
+    featureType: SimpleFeatureType,
+    attributes: IntermediaryValues): SimpleFeature = {
+
+    val id = new FeatureIdImpl(featureId.toString)
+
+    val nonIds = attributes.filter { v =>
+      v match {
+        case (_: SoQLID, _) => false
+        case _ => true
```

Comment:
these are intermediary:value tuples, so i think the case is more expressive than
 `!_._1.isInstanceOf[SoQLID]` which just seems a little silly, but i'm happy to change it to that